File: Java/2331. Evaluate Boolean Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean helper(TreeNode node){
        if(node.val == 0) return false;
        if(node.val == 1) return true;
        if(node.val == 2) return helper(node.left) || helper(node.right);
        if(node.val == 3) return helper(node.left) && helper(node.right);
        return true;
    }
    public boolean evaluateTree(TreeNode root) {
        return helper(root);
    }
}

File: Python/2331. Evaluate Boolean Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def evaluateTree(self, root: Optional[TreeNode]) -> bool:
        def helper(node):
            if node.val == 0: return False
            if node.val == 1: return True
            if node.val == 2: return helper(node.left) or helper(node.right)
            if node.val == 3: return helper(node.left) and helper(node.right)
        
        return helper(root)
